src/commands/insertcolumncommand.js

```diff
@@ -70,15 +70,35 @@ export default class InsertColumnCommand extends Command {
 		const editor = this.editor;
 		const selection = editor.model.document.selection;
 		const tableUtils = editor.plugins.get( 'TableUtils' );
+		const contentLanguageDirection = editor.locale.contentLanguageDirection;
 
 		const firstPosition = selection.getFirstPosition();
 
 		const tableCell = findAncestor( 'tableCell', firstPosition );
 		const table = tableCell.parent.parent;
 
-		const { column } = tableUtils.getCellLocation( tableCell );
-		const insertAt = this.order === 'right' ? column + 1 : column;
+		let { column } = tableUtils.getCellLocation( tableCell );
+		const isOrderRight = this.order === 'right';
+		const isContentLtr = contentLanguageDirection === 'ltr';
 
-		tableUtils.insertColumns( table, { columns: 1, at: insertAt } );
+		// In RTL content, the table is (visually) mirrored horizontally. Columns logically "before" are
+		// displayed on the right–side and vice–versa. So for the UI/UX to make sense, commands must
+		// work the other way around. It is confusing and it is easy to get it wrong.
+		//
+		//                              ┌──────────────────────────────┐
+		//                              │            order             │
+		//                              ├───────────────┰──────────────┤
+		//                              │    'right'    │    'left'    │
+		//          ┌───────────┰───────┼───────────────┼──────────────┤
+		//          │   content │ 'ltr' │   column+1    │    column    │
+		//          │  language │───────┼───────────────┼──────────────┤
+		//          │ direction │ 'rtl' │    column     │   column+1   │
+		//          └───────────┴───────┴───────────────┴──────────────┘
+		//
+		if ( ( isOrderRight && isContentLtr ) || ( !isOrderRight && !isContentLtr ) ) {
```

Comment:
This gets complicated with RTL. And I wonder if we shouldn't put emphasis on the UI for dictating "left"/"right" and have "before"/"after" in the model?

This way the command will accept "before" / "after" params from the UI and will execute them in the model order (as in arrays). The UI would switch command param depending on the content to match the visual aspect of it.

Reply:
In other words, the model manipulation should be simpler. It should focus on data direction as in after/before. The visual aspect of the editor, at least in here, could be done on the UI side of things.

This makes sense for me that